src/services/BackendService/BackendService.ts

```diff
@@ -92,10 +102,47 @@ export class BackendService implements BackendInterface {
     };
   }
 
+  filterTEKs = (contagiousDateInfo?: ContagiousDateInfo) => {
+    return (key: TemporaryExposureKey) => {
+      // This function filters out TEKs that were generated before the user was contagious.
+      // rollingStartIntervalNumber = A number describing when a key starts. It is equal to
+      // startTimeOfKeySinceEpochInSecs / (60 * 10).
+      // rollingPeriod = A number describing how long a key is valid. It is expressed in
+      // increments of 10 minutes (e.g. 144 for 24 hours).
+      // source: https://developers.google.com/android/reference/com/google/android/gms/nearby/exposurenotification/TemporaryExposureKey
+      if (!contagiousDateInfo || contagiousDateInfo.dateType === 'noDate' || !contagiousDateInfo.dateString) {
+        return true;
+      }
+      // the following is a bit hacky - maybe we should pass the date as a Date, not a string
+      const dateParts = contagiousDateInfo.dateString.split('-');
+      const providedDate = new Date(Number(dateParts[0]), Number(dateParts[1]) - 1, Number(dateParts[2]));
```

Comment:
No need to use 'split' here; passing in a string to `new Date()` or `Date.parse()` should return a Date object.